quesma/quesma/router_test.go

```diff
@@ -181,3 +183,97 @@ func Test_matchedAgainstBulkBody(t *testing.T) {
 		})
 	}
 }
+
+func Test_combineSourcesFromElasticWithRegistry(t *testing.T) {
```

Comment:
`router_test.go` is probably not the best place for this test, similarly for `combineSourcesFromElasticWithRegistry` in `router.go`.

Should I move it somewhere else? (where?)

Reply:
I'd create a dedicated file for `_resolve`-related tests

Reply:
OK, what about the implementation itself (currently in `router.go`)? Should it be in `resolve.go` or maybe somewhere in `schema.go`, `registry.go`?

Reply:
Ideally, the functionality should be pulled into a dedicated package and live alongside those:
<img width="321" alt="image" src="https://github.com/user-attachments/assets/b81bb5cd-e3e3-4718-892b-699d7c6696a3">

The router should merely just use the exported functionality of a new package

Reply:
Moved the code and the test to `functionality/resolve` package.